src/lib/native.ts

```diff
@@ -0,0 +1,90 @@
+import * as dialog from "@tauri-apps/api/dialog"
+import * as fs from "@tauri-apps/api/fs"
+import * as path from "@tauri-apps/api/path"
+import * as shell from "@tauri-apps/api/shell"
+import * as app from "@tauri-apps/api/app"
+import * as os from "@tauri-apps/api/os"
+import * as window from "@tauri-apps/api/window"
+import * as tProcess from "@tauri-apps/api/process"
+// import { invokeTauriCommand } from "@tauri-apps/api/"
+import { invoke } from '@tauri-apps/api/tauri'
+import macAL from "src/lib/autoLaunch/macAutoLaunch"
+import winAL from "src/lib/autoLaunch/nullAutoLaunch"
+import linAL from "src/lib/autoLaunch/nullAutoLaunch"
+import nullAL from "src/lib/autoLaunch/nullAutoLaunch"
+import { ChildReturnData } from "./types"
+type osAL = typeof macAL | typeof winAL | typeof linAL | typeof nullAL
+const tauri = { app, dialog, fs, path, invoke, shell, os, window, process: tProcess }
+export interface DriveStats {
+  freeBytes: number
+  totalBytes: number
+}
+export interface TauriDriveStats {
+  free_bytes: number
+  total_bytes: number
+}
+export class AutoLauncher {
+  protected autoLauncher: osAL = nullAL
+  appName: string = 'app'
+  appPath: string = ''
+  enabled: boolean = false
+  async isEnabled(): Promise<boolean> {
+    const result = await this.autoLauncher.isEnabled(this.appName)
+    this.enabled = result
+    return result
+  }
+  async enable(): Promise<void | ChildReturnData> {
+    const child = await this.autoLauncher.enable({ appName: this.appName, appPath: this.appPath, hidden: false })
+    return child
+  }
+  async disable(): Promise<void | ChildReturnData> {
+    const child = this.autoLauncher.disable(this.appName)
+    return child
+  }
+  async init() {
+    this.appName = process.env.DEV ? "app" : (await tauri.app.getName()).toString()
+    const os = await tauri.os.type()
+    this.appPath = await tauri.invoke('get_this_binary') as string
+    console.log('get_this_binary', this.appPath);
+    console.log(this.appPath);
+    if (os == 'Darwin') this.autoLauncher = macAL
+    else if (os == 'Windows_NT') this.autoLauncher = winAL
+    else this.autoLauncher = linAL
+    this.enabled = (await this.isEnabled())
+  }
+}
+
+export async function execString(executable: string, args: string[] | string) {
```

Comment:
This function returns `boolean` and while it isn't specified, nothing complains. This should fail at compile time complaining about wrong types, the correct return type of this function is `Promise<boolean>`. The same comment applies to other functions in this file. Without such checks, we essentially don't have TypeScript and can return random things from random functions with wrong signatures.

Let me know if this is already taken care of by eslint in the other PR (I was using tslint for similar purpose before, but it is discontinued, I'm sure there are similar options for eslint).

Reply:
I agree that many functions in this file could be cleaned up. There is no linting or compile errors so I'm not sure what setting we need to have to formally mark these as errors.  I think most issues resolved in eslint branch #9